### PR TITLE
Update Compatibility with Elasticsearch 7

### PR DESCRIPTION
**Pull request by tmotyl**
* bump Elastica version
* migrate away from Type\Mapping & Index has no getType method any more see:
https://github.com/ruflin/Elastica/pull/1666
* Use real index name instead of alias when adding new alias.
* wrap index configuration in "settings" key to conform with the new PUT syntax 
https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#create-index-settings
https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-modules-settings

Tested with ES 7.11

Important links:
https://www.elastic.co/guide/en/elasticsearch/reference/current/removal-of-types.html

### CODE REVIEW

**Comment by Lefaux** [comment]
I don't have a checked out version at hand right now, but I suggest the readme.md has the new compatibility updated.

**Comment by tmotyl** [comment]
@Lefaux lets first merge 
https://github.com/TYPO3GmbH/elasticorn/pull/19 and https://github.com/TYPO3GmbH/elasticorn/pull/18
As they are useful for older versions too I believe 

In this PR I'm trying to fix some tests, so please don't merge it yet.

File: Tests/Acceptance/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -106,11 +107,11 @@ public function iHaveSetupMappingsAndDataForIndex(string $indexName)
 
         $baseIndex = $client->getIndex('footest');
         $mapping = $baseIndex->getMapping();
-        foreach ($mapping as $documentType => $properties) {
```
Comment (tmotyl) [inline Tests/Acceptance/features/bootstrap/FeatureContext.php:109]: new ES7 doesn't have types any more.

**Comment by tmotyl** [comment]
Current status:
- "it works for me" ;) I'm able to create index with languages from config, yay!
- there is one unit test failing - Elastica has added return types definitions, so mocking needs to be updated
- As ES7 doesn't have a concept of "types" any more. Currently the behat tests are testing cases with multiple types, so they fail.
A decision about how to  proceed has to be made - should types be substituted with some other mechanism in elasticorn? or should we just simplify tests to cover a single type?

**Comment by susannemoog** [comment]
As I'm not allowed to update this PR, I created a new branch in this repo with your changes, rebased to current master. I additionally added the basic CI setup that has been missing in this project as we did not do much here for the last few months/years and updated the dependencies to maintained and secure versions. Please check https://github.com/TYPO3GmbH/elasticorn/tree/es_v7 and see if you can integrate that with your changes. 
The acceptance tests are still not running in CI but at least CGL and unit tests do.

As for the other changes: I merged and released the other two patches and build new phars for that. For Elasticsearch 7: I would declare the change as breaking and suggest to remove the document types and use separate indices as proposed by elasticsearch themselves.

**Comment by tmotyl** [comment]
Thanks @susannemoog 
I've tested the branch es_v7 from this repo and creating indexes works with ES 7.8 and 7.11.

**Pull request merged**